src/components/Auth/RegisterForm.vue

```diff
@@ -0,0 +1,398 @@
+<script>
+import { defineComponent } from 'vue';
+
+import { mapActions } from 'pinia';
+import { useAuthStore } from '@/store/auth';
+
+import useVuelidate from '@vuelidate/core';
+import { required, email, helpers } from '@vuelidate/validators';
+import {
+  hasNumber,
+  hasLetter,
+  least8Chars,
+  min18Years,
+  privPolicyChecked
+} from '@/helpers/validations';
+
+import Datepicker from '@vuepic/vue-datepicker';
+
+import BaseButton from '@/components/global/BaseButton.vue';
+import BaseTextInput from '@/components/global/BaseTextInput.vue';
+import BaseCheckbox from '@/components/global/BaseCheckbox.vue';
+
+import AuthHeader from '@/components/Auth/AuthHeader.vue';
+
+export default defineComponent({
+  name: 'RegisterForm',
+  components: {
+    AuthHeader,
+    BaseButton,
+    BaseTextInput,
+    BaseCheckbox,
+    Datepicker
+  },
+  data() {
+    return {
+      v$: useVuelidate(),
+      activeStep: 1,
+      email: '',
+      password: '',
+      firstName: '',
+      lastName: '',
+      dateOfBirth: '',
+      privPolicy: false,
+      passwordHints: [
+        { id: 1, message: 'At least 8 characters' },
+        { id: 2, message: 'At least one letter' },
+        { id: 3, message: 'At least one digit' }
+      ]
+    };
+  },
+  computed: {},
+  methods: {
+    ...mapActions(useAuthStore, ['register']),
+    nextStep() {
+      if (!this.v$.email.$invalid && !this.v$.password.$invalid) {
+        this.activeStep < 2 ? this.activeStep++ : this.activeStep;
+      } else alert('Some form fields must be empty!');
+    },
+    async onSubmit() {
+      this.v$.$validate();
+      if (!this.v$.$invalid && this.privPolicy) {
+        try {
+          await this.register({
+            email: this.email,
+            password: this.password,
+            dateOfBirth: this.dateOfBirth,
+            firstName: this.firstName,
+            lastName: this.lastName
+          });
+
+          this.$router.push({ name: 'Home' });
+        } catch (error) {
+          alert(error);
+          console.error(error);
+        }
+      } else
+        alert(
+          'Some form fields must be empty or invalid, please check the hints!'
+        );
```

Comment:
As a matter of good practice - never skimp on the brackets and keep the code consistent 😄
 
```suggestion
      } else {
        alert(
          'Some form fields must be empty or invalid, please check the hints!'
        );
      }
```

Reply:
You got me there.. It's a bad habit, but I'll try to remember to keep things well bracketed :ok_hand: